cuda/private/defs.bzl

```diff
@@ -0,0 +1,16 @@
+"""private defs"""
+
+def requires_cuda():
+    """Returns constraint_setting that is satisfied if:
+
+    * CUDA is enabled and
+    * CUDA toolchain is found.
+
+    Add to 'target_compatible_with' attribute to mark a target incompatible when
+    @rules_cuda//cuda:is_enabled_and_cuda_found is not set. Incompatible targets are excluded
+    from bazel target wildcards and fail to build if requested explicitly.
+    """
+    return select({
+        "@rules_cuda//cuda:is_enabled_and_cuda_found": [],
+        "//conditions:default": ["@platforms//:incompatible"],
```

Comment:
I am wondering if it is possible to alias `incompatible`, so that when `is_enabled` is false, it can somehow reports `incompatible` because the rule is disabled and when `is_cuda_found` is false, it can also report the toolchain is not configure.

Otherwise, when `incompatible` occurs, you still need to dig into it to figure out why it happens, and somehow make it a little bit worse because previous they are two distinct error, now they are just `incompatible`.

Reply:
Refactored.